src/map/ai/controllers/mob_controller.cpp

```diff
@@ -620,8 +620,9 @@ void CMobController::Move()
         }
     }
 
-    bool  move         = PMob->PAI->PathFind->IsFollowingPath();
-    float attack_range = PMob->GetMeleeRange();
+    bool  move            = PMob->PAI->PathFind->IsFollowingPath();
+    float attack_range    = PMob->GetMeleeRange();
+    float closureDistance = PMob->getMobMod(MOBMOD_ENCROACH_TARGET) ? attack_range - 3.7f : attack_range - 0.2f;
```

Comment:
why not use the value of the mod so that this mod can be reused at variable distances? And call it `MOBMOD_DISTANCE` 

if zero, use the default, take mobmod and convert to float (that 0.2f is in yalms btw)

I'd say just modify clarets attack range at the db level, but that would adjust attack range players can hit back as well.

Reply:
> mobMods are integer values

conversion is a trivial casting

> The ideal way to make this variable would be to consider mob size (model size) into the equation 

we already do this in the attack range, which is based on the family in the mobs family currently tho I would like to move this to someplace that can be more flexible (and check the bit flags for alternate sizes on the same model, as abysea mobs do to spawn in fixed sizes like most others)

Reply:
True, conversion is cheap.
I can add the configurablility - I assume somone will make use of it someday, hopefully not a YAGNI

On the topic of mob sizes - one of the biggest pains for this on wings was blue magic - havent looked into that on LSB yet.  I created a custom hack which would consider mob size into the distance calc, otherwise blu's had to clip into larger mobs to be in range.

Reply:
really the hold up for us is that the size bits don't have universal behavior among all entities that have that mask so we already know we're going to refactor the mask the size bits sit in, so anything we do there would wind up getting redone :disappointed: 

as far as YAGNI, even if we never find another use case, I think it's a good method for the use here while opening up future use even if future use turns out to be some random server's custom :+1: